File: Android-Firebase-Notification-SourceCode/app/src/main/java/com/hellohasan/android_firebase_notification/util/Preferences.java
package com.hellohasan.android_firebase_notification.util;

import android.content.Context;
import android.content.SharedPreferences;


public class Preferences {

    public static final String VERSION_CODE = "version_code";

    private static SharedPreferences sharedPreferences;
    private static SharedPreferences.Editor editor;

    private static Preferences preferences = new Preferences();

    private Preferences(){}


    public static Preferences getInstance(Context context){
        sharedPreferences = context.getSharedPreferences("sharedPreferences_data", Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        return preferences;
    }


    public long getVersionCode(){
        return sharedPreferences.getLong(VERSION_CODE, -1);
    }

    public void setVersionCode(long versionCode){
        editor.putLong(VERSION_CODE, versionCode);
        editor.apply();
        editor.commit();
    }

}

File: AndroidFirebaseNotification/app/src/main/java/com/hellohasan/android_firebase_notification/utils/Preferences.kt
package com.hellohasan.android_firebase_notification.utils

import android.content.Context
import android.content.SharedPreferences


class Preferences private constructor() {


    var versionCode: Long
        get() = sharedPreferences!!.getLong(VERSION_CODE, -1)
        set(versionCode) {
            editor?.putLong(VERSION_CODE, versionCode)
            editor?.apply()
            editor?.commit()
        }

    companion object {

        val VERSION_CODE = "version_code"

        private var sharedPreferences: SharedPreferences? = null
        private var editor: SharedPreferences.Editor? = null

        private val preferences = Preferences()


        fun getInstance(context: Context): Preferences {
            sharedPreferences =
                context.getSharedPreferences("sharedPreferences_data", Context.MODE_PRIVATE)
            editor = sharedPreferences?.edit()
            return preferences
        }
    }

}